private/bufpkg/bufstate/global_state.go

```diff
@@ -67,21 +67,3 @@ func WriteGlobalState(writer io.WriteCloser, globalState *statev1alpha1.GlobalSt
 	}
 	return nil
 }
-
-// validateGlobalState checks there are no repeated modules or empty values in names or references.
-func validateGlobalState(s *statev1alpha1.GlobalState) error {
```

Comment:
This validation and the one in `module_state.go` were moved to be protovalidate rules.

Reply:
The goal is to not have the backend depend on this module, yes, that's why we embed the validation in protovalidate rules, so we can protovalidate from the backend as well.